sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataHolder.java

```diff
@@ -36,7 +38,8 @@
   private final long isAvailable;
   private final long isRealtime;
   private final String segmentId;
-  private final long numReplicas;
+  //segmentId -> set of servers that contain the segment
+  private final Map<String, Set<String>> segmentServerMap;
```

Comment:
This map seems to always have just one entry, isn't it?

Reply:
Yeah, you are right, should have used a `Pair` instead.

Reply:
Won't `lhs` in that Pair always be equal to `segmentId` field? In other words shouldn't it be just `Set<String> servers` field?

Reply:
ah, yeah. thanks, will change with another upcoming PR in this area.